router/http/src/main/scala/io/buoyant/router/http/PathIdentifier.scala

```diff
@@ -8,12 +8,16 @@ import io.buoyant.router.RoutingFactory
 case class PathIdentifier(
   prefix: Path,
   segments: Int = 1,
+  consume: Boolean = false,
   baseDtab: () => Dtab = () => Dtab.base
 ) extends RoutingFactory.Identifier[http.Request] {
 
   def apply(req: http.Request): Future[(Dst, http.Request)] =
     Path.read(req.path) match {
       case path if path.size >= segments =>
+        if (consume) {
+          req.uri = req.uri.split("/").drop(segments + 1).mkString("/", "/", "")
```

Comment:
TIOLI: there are probably ways of doing this with fewer copies (e.g. `req.uri.drop(captured.show.length)`) -- probably not worth optimizing yet